src/Sentry.AspNetCore/SentryMiddleware.cs

```diff
@@ -104,7 +104,9 @@ public async Task InvokeAsync(HttpContext context)
                     // event creation will be sent to Sentry
 
                     scope.OnEvaluating += (_, __) => PopulateScope(context, scope);
+                    scope.Transaction.StartTimestamp = DateTimeOffset.Now;
```

Comment:
Here we shoudl be doing:

@HazAT @rhcarvalho @untitaker @maciejwalkowiak does the pseudo code below make sense?


```
var transactionName = GetName(); // we get the name somewhere because it's added today to the event.transaction
var transaction = hub.StartTransaction(transactionName, new SamplingContext())

// Ideally we'd add to some data bag of the context, so we can get from it in different parts of the framework like some MVC Filters or something, and are able to create more spans from it, without having to rely on `AsyncLocal`:
context.SomeDataBag["SentryTransaction"] = transaction;

// ALternatively we rely on AsyncLocal, or both:
Hub.ConfigureScope(s => s.SetTransaction(transaction));

try {
var middlewareSpan = transaction.startSpan("middleware") (or startChild or something);
await middleware();
middlewareSpan.end();
transaction.status(context.Response.Status.ToSentryStatus());
} catch {
transaction.status(exception);
}
finally {
transaction.finish()
}

```

Reply:
`var transaction = hub.StartTransaction(transactionName, new SamplingContext())` should rather take `CustomSamplingContext` instead of `SamplingContext`.

`Hub.ConfigureScope(s => s.SetTransaction(transaction));` in Java once you start transaction it's set on the scoep automatically.

Other than that it does make sense.

Reply:
If it can serve as inspiration, here's the equivalent middleware in Go: https://github.com/getsentry/sentry-go/blob/cca482cffbf600853b2f3abda730f9d5a97ee9aa/http/sentryhttp.go#L80-L101

1. The transaction name is stored in the `Scope` (as there was already a slot for it there, and Tracing support hijacks that event field)
2. The "some data bag" idea is represented by Go's `Context` type. Spans are stored there. There's no transaction type per-se. A transaction is a wire representation of a span tree (in fact, just a special type of Event).
3. ⚠️ Bruno your code creates both a transaction AND a child span. But if you represent the transaction as a thing on its own, it is already a span. So in your example there's one too many nesting levels. Either the middleware starts a transaction or a child span on an existing transaction. I don't like code needing to check whether it needs to call `StartTransaction` or `StartSpan`, that's why, among other reasons, there's only `StartSpan` in Go.
4. Apart from setting the span status, the Python SDK also sets a tag with the HTTP status. We should do that too.
[integrations/aiohttp.py#L112](https://github.com/getsentry/sentry-python/blob/ab3da0809d6c2c32adfa63917af03a58cd498fd3/sentry_sdk/integrations/aiohttp.py#L112)
[tracing.py#L319-L350](https://github.com/getsentry/sentry-python/blob/f6f3525f8812f60911573a8b7f71807ecf2e1052/sentry_sdk/tracing.py#L319-L350)
5. When starting a transaction/span, we pass data that modify the transaction/span (the so called "TransactionContext" and "SpanContext" in JS). The "SamplingContext" is something else, it originates in the SDK and is passed to a TracesSampler.

Reply:
One point to add:

6. The `op` for HTTP server integrations should be `"http.server"`, so we all follow a consistent naming pattern -- unless there's good reason to deviate from that.

Reply:
> The op for HTTP server integrations should be "http.server", so we all follow a consistent naming pattern -- unless there's good reason to deviate from that.

Is this documented somewhere?

Reply:
Possibly not :/
It started off prior industry specs and we use it for Python as in [integrations/aiohttp.py#L92](https://github.com/getsentry/sentry-python/blob/ab3da0809d6c2c32adfa63917af03a58cd498fd3/sentry_sdk/integrations/aiohttp.py#L92)